workflow/rules/stats.smk

```diff
@@ -0,0 +1,30 @@
+rule samtools_flagstat:
+    input:
+        "results/{path_and_file_name}.bam"
+    output:
+        "results/{path_and_file_name}.flagstat"
+    wrapper:
+        "0.60.0/bio/samtools/flagstat"
```

Comment:
Maybe we could make the wildcards slightly more finegrained and thus keep the bam file names simple, while adding the necessary specs into the stats file names. This would require limiting any of the bam file outputs to a maximum of one further folder-level underneath `results`, so e.g. `results/merged/dedup` would become `results/dedup`, which should be fine, as we only have one dedup step after all (we only initially thought, this would happen twice).
```
    input:
        "results/{step}/{sample}.bam"
    output:
        "results/{step,[^./]+}/{sample}.{step}.flagstat"
```
Does this make sense? Could you try this out?

Reply:
That's a great idea, thanks! It's going great! But with stats.smk and utils.smk it is not enough to define only {sample} as wildcard, because then the globally defined wildcard is accessed. But for the statistics after bwa_mem a {sample}-{unit} combination is needed. So I redefined the wildcars as {samples_units} and than in the common.smk the exact names can be adjusted. Unfortunately the sorting in General Statistics in multiqc.html is now based on the sample name and not on the steps, so it looks a bit confusing. On the other hand the grouping by samples could be advantageous.